tests/unit-ext/test-ext-module-canonical.c

```diff
@@ -0,0 +1,93 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string.h>
+
+#include "jerryscript.h"
+#include "test-common.h"
+#include "jerryscript-ext/module.h"
+
+static jerry_value_t
+get_canonical_name (const jerry_value_t name)
+{
+  jerry_size_t name_size = jerry_get_string_size (name);
+  jerry_char_t name_string[name_size + 1];
+  jerry_string_to_char_buffer (name, name_string, name_size);
+  name_string[name_size] = 0;
+
+  return (!strcmp ((char *) name_string, "batman") ? jerry_acquire_value (name):
+         (!strcmp ((char *) name_string, "bruce-wayne") ? jerry_create_string ((jerry_char_t *) "batman"):
+         jerry_create_undefined ()));
+} /* get_canonical_name */
+
+static bool
+resolve (const jerry_value_t canonical_name, jerry_value_t *result)
+{
+  jerry_size_t name_size = jerry_get_string_size (canonical_name);
+  jerry_char_t name_string[name_size];
+  jerry_string_to_char_buffer (canonical_name, name_string, name_size);
+  if (!strncmp ((char *) name_string, "batman", 6))
+  {
+    *result = jerry_create_object ();
+    return true;
+  }
+  return false;
+} /* resolve */
+
+static const jerryx_module_resolver_t canonical_test =
+{
+  get_canonical_name,
+  resolve
+};
+
+#define TEST_VALUE 95.0
+
+int
+main (int argc, char **argv)
+{
+  (void) argc;
+  (void) argv;
+
+  const jerryx_module_resolver_t *resolver = &canonical_test;
+
+  jerry_init (JERRY_INIT_EMPTY);
+
+  jerry_value_t batman = jerry_create_string ((jerry_char_t *) "batman");
+  jerry_value_t bruce_wayne = jerry_create_string ((jerry_char_t *) "bruce-wayne");
```

Comment:
I hope these are not copyrighted.

Reply:
Good point. I'll use [Liberator](http://pdsh.wikia.com/wiki/Liberator), which is public domain.

Reply:
They are still, however, "use at your own risk" characters. - this doesn't sound entirely public

Reply:
*sigh* Alice and Bob it is.